adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -251,6 +256,18 @@ public static AuthenticationResult processUIResponseParams(Map<String, String> r
 
         } else if (response.containsKey(AuthenticationConstants.OAuth2.CODE)) {
             result = new AuthenticationResult(response.get(AuthenticationConstants.OAuth2.CODE));
+            final String cloudInstanceHostName = response.get(AuthenticationConstants.OAuth2.CLOUD_INSTANCE_HOST_NAME);
+            if (!StringExtensions.isNullOrBlank(cloudInstanceHostName)) {
+
+                final URL authorityUrl = StringExtensions.getUrl(mRequest.getAuthority());
+                final StringBuilder newAuthorityUrlString = new StringBuilder(authorityUrl.getProtocol())
+                        .append("://")
+                        .append(cloudInstanceHostName)
+                        .append(authorityUrl.getPath());
+
+                setTokenEndpoint(newAuthorityUrlString.toString() + DEFAULT_TOKEN_ENDPOINT);
+                result.setAuthority(newAuthorityUrlString.toString());
```

Comment:
>result.setAuthority(newAuthorityUrlString.toString()); [](start = 16, length = 54)

Can you make setAuthority at request level?

Reply:
Sorry, I don't understand, we need to return the authority to the called, what will setting it to the request achieve?